Fruitties/iosApp/iosApp/ContentView.swift

```diff
@@ -66,38 +69,36 @@ struct FruittieView: View {
     }
 }
 
-class UIModel: ObservableObject {
-    let dataRepository : DataRepository
-    init(dataRepository: DataRepository) {
-        self.dataRepository = dataRepository
+@Observable class UIModel {
+    private let mainViewModel: MainViewModel
+    init(mainViewModel: MainViewModel) {
+        self.mainViewModel = mainViewModel
     }
-    @Published
     private(set) var fruitties: [Fruittie] = []
-    @Published
-    private(set) var cart: Cart = Cart(items: [])
+    private(set) var cartDetails: [CartItemDetails] = []
 
     @MainActor
-    func observeDatabase() async {
-        for await fruitties in dataRepository.getData() {
-            self.fruitties = fruitties
+    func observeHomeUiState() async {
```

Comment:
Thanks! I updated a few variables. I did not update ContentView.uiModel because it would conflict with the class name, `UIModel` -- I think that matches with the lowerCamelCase style.